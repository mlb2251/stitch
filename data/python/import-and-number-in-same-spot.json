[
  "(Module (/seq (Expr (Constant s_x None)) (Import (list (alias &os:0 None))) (Import (list (alias &sys:0 None))) (Expr (Attribute (Name &os:0 Load) s_environ Load))) nil)",
  "(Module (/seq (Assign (list (Name &i:0 Store)) (Constant s_x None) None) (Import (list (alias &copy:0 None))) (Import (list (alias &re:0 None)))) nil)"
]