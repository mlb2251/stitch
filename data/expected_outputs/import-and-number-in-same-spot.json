{
  "cmd": "/home/kavi/mit/stitch/target/debug/deps/integration_tests-96f63b0742106ff9 symbols_basic",
  "args": {
    "iterations": 1,
    "abstraction_prefix": "fn_",
    "previous_abstractions": 0,
    "shuffle": false,
    "truncate": null,
    "no_opt": false,
    "silent": false,
    "verbose_rewrite": false,
    "step": {
      "max_arity": 0,
      "threads": 1,
      "no_stats": false,
      "batch": 1,
      "dynamic_batch": false,
      "eta_long": false,
      "no_curried_metavars": false,
      "no_curried_bodies": false,
      "inv_candidates": 1,
      "hole_choice": "DepthFirst",
      "cost": {
        "cost_lam": 1,
        "cost_app": 1,
        "cost_var": 100,
        "cost_ivar": 100,
        "cost_prim_default": 100,
        "cost_prim": "{}"
      },
      "no_mismatch_check": false,
      "follow": null,
      "follow_prune": false,
      "verbose_worklist": false,
      "verbose_best": false,
      "print_stats": 0,
      "show_rewritten": false,
      "rewritten_dreamcoder": false,
      "rewritten_intermediates": false,
      "inv_arg_cap": false,
      "allow_single_task": false,
      "no_opt_single_use": false,
      "no_opt_upper_bound": false,
      "no_opt_force_multiuse": false,
      "no_opt_useless_abstract": false,
      "no_opt_arity_zero": false,
      "no_other_util": false,
      "structure_penalty": 1.0,
      "rewrite_check": false,
      "utility_by_rewrite": false,
      "dreamcoder_comparison": false,
      "quiet": false,
      "fused_lambda_tags": {
        "tags": null
      },
      "tdfa": {
        "tdfa_json_path": "test_data/dfa.json",
        "tdfa_root": "M",
        "valid_metavars": "[\"S\",\"E\",\"seqS\"]",
        "valid_roots": "[\"S\",\"E\",\"seqS\"]",
        "tdfa_non_eta_long_states": "{\"seqS\":\"S\"}"
      },
      "symvar": {
        "symvar_prefix": "&"
      }
    }
  },
  "original_cost": 9694,
  "final_cost": 9290,
  "compression_ratio": 1.043487621097955,
  "num_abstractions": 1,
  "original": [
    "(Module (/seq (ImportFrom s_math (list (alias g_sqrt None)) i0) (Expr (BinOp (BinOp (Call (Name g_sqrt Load) (list (_starred_content (Constant i123 None))) nil) Add (Call (Name g_sqrt Load) (list (_starred_content (Constant i2 None))) nil)) Add (Constant i7 None)))) nil)",
    "(Module (/seq (Assign (list (Name &f:0 Store)) (Lambda (arguments nil (list (arg &x:1 None None)) None nil nil None nil) (BinOp (Name &x:1 Load) Pow (Constant f0.5 None))) None) (Expr (BinOp (BinOp (Call (Name &f:0 Load) (list (_starred_content (Constant i123 None))) nil) Add (Call (Name &f:0 Load) (list (_starred_content (Constant i2 None))) nil)) Add (Constant i7 None)))) nil)"
  ],
  "rewritten": [
    "(Module (/seq (ImportFrom s_math (list (alias g_sqrt None)) i0) (Expr (BinOp (BinOp (Call (Name g_sqrt Load) (list (_starred_content fn_0)) nil) Add (Call (Name g_sqrt Load) (list (_starred_content (Constant i2 None))) nil)) Add (Constant i7 None)))) nil)",
    "(Module (/seq (Assign (list (Name &f:0 Store)) (Lambda (arguments nil (list (arg &x:1 None None)) None nil nil None nil) (BinOp (Name &x:1 Load) Pow (Constant f0.5 None))) None) (Expr (BinOp (BinOp (Call (Name &f:0 Load) (list (_starred_content fn_0)) nil) Add (Call (Name &f:0 Load) (list (_starred_content (Constant i2 None))) nil)) Add (Constant i7 None)))) nil)"
  ],
  "rewritten_dreamcoder": null,
  "abstractions": [
    {
      "body": "(Constant i123 None)",
      "dreamcoder": "#(Constant i123 None)",
      "arity": 0,
      "name": "fn_0",
      "utility": 102,
      "final_cost": 9290,
      "compression_ratio": 1.043487621097955,
      "cumulative_compression_ratio": 1.043487621097955,
      "num_uses": 2,
      "rewritten": null,
      "rewritten_dreamcoder": null,
      "uses": [
        {
          "fn_0 ": "(Constant i123 None)"
        }
      ],
      "dc_comparison_millis": null,
      "tdfa_annotation": {
        "root_state": "E",
        "metavariable_states": []
      }
    }
  ]
}